src/HttpExtractor.php

```diff
@@ -21,7 +25,23 @@ public function __construct(
 
     public function extract(UriInterface $httpSource, string $filesystemDestination): void
     {
-        $this->client->get($httpSource, ['sink' => $filesystemDestination]);
+        try {
+            $this->client->get($httpSource, ['sink' => $filesystemDestination]);
+        } catch (ClientException|ServerException $e) {
+            throw new UserException(sprintf(
+                'Server returned HTTP %s for "%s"',
+                $e->getCode(),
+                (string)$httpSource
+            ), 0, $e);
+        } catch (ConnectException $e) {
+            if ($e->getHandlerContext()['errno'] !== CURLE_COULDNT_RESOLVE_HOST) {
```

Comment:
ja vim, ze uz jsem se na to jednou dival, ale proc je tu vlastne ten CURLE_COULDNT_RESOLVE_HOST ? Jakoze `CURLE_COULDNT_RESOLVE_HOST` konvertujeme na user error, ale osatni ne? No ale proc?

Reply:
Já to beru tak, že je to dostatečně jasná chyba na to, abysme ji vyhodili jako chybu uživatele. Těch možných chyb je hromada: http://docs.factorcode.org/content/vocab-curl.ffi.html a ne všechny jsou chyby uživatele. Viděl bych to tak, že pokud se často budou objevovat nějaké další typy, tak je tam můžeme přidávat.

Reply:
Ale ve výsledku to můžeme odstranit, ale nevím, jak to vhodně udělat uživatelsky. Namapovat na každý možný CURLE nějakou message? Nebo prostě vyhodit "Chyba spojení:" + to co tam má curl?

Reply:
ok, v tom pripade bych tam pridal tyto https://developers.keboola.com/extend/generic-extractor/configuration/api/#configuration  ktere se objevuji pomerne casto

Reply:
no otazka spis smerovala na to, jestli ty ostatni curl chyby maji byt application errory

Reply:
Ale není rozdíl mezi "resumable" a "user-caused"?

Reply:
no retry zadnej nedelame (mohli bysme), takze neni, protoze ho musi udelat uzivatel - cili resumable = user-caused v tuto chvili

kazdopadne usererror je to s cim muze end-user neco delat
application error je to, co musime resit my

Reply:
A máme ještě nějaké další možné chování? Něco jako `RetryableException` nebo `TransientException` (resp. `exit(?)`)? Mám za to, že chyba uživatele je jen to, co může uživatel opravit. 

Samozřejmě by se dalo zavést něco jako ten exponencial backoff, ale to je spíš nová feature, ne?

Reply:
Úplně jinak - správná otázka pro mě je: "Proč je špatně, že jsou to teď aplikační chyby?" Ta jediná mi totiž pomůže pochopit proč to chceš ;)

Reply:
Ha, sorry, je to tady takové asynchronní. Už to chápu.

Reply:
nemame (ale nejakej uvahy na to tema mame https://github.com/keboola/connection/issues/799 :) 

uzivatel to muze opravit tak, ze to proste spusti znovu (tj, musi videt, ze resenim je spustit to znovu) a udela si carku, ze dodavatel je nespolehlivej 

jo, retry bych bral jako novou ficuru

Reply:
Pushnuto. Takhle OK?